core/types/row_consumption.go

```diff
@@ -12,4 +14,38 @@ type SubCircuitRowUsage struct {
 	RowNumber uint64 `json:"row_number" gencodec:"required"`
 }
 
+// RowConsumptionLimit is the max number of row we support per subcircuit
+const RowConsumptionLimit = 1_000_000
+
 type RowConsumption []SubCircuitRowUsage
+
+// IsOverflown returns if any subcircuits are overflown
+func (rc RowConsumption) IsOverflown() bool {
+	return slices.ContainsFunc(rc, func(scru SubCircuitRowUsage) bool {
+		return scru.RowNumber > RowConsumptionLimit
+	})
+}
+
+// Difference returns rc - other
+// Assumes that rc > other for all subcircuits
```

Comment:
i think `rc > other` won't always hold?

Reply:
Caller will have to make sure that it makes sense to call Difference with the row consumptions they have.
Intended usecase for this is something like this;

```
rc1 := applyTxAndCalculateAccumulatedRowConsumption(block, tx1)
rc2 := applyTxAndCalculateAccumulatedRowConsumption(block, tx2) // (rc2 also includes rc1)

rowConsumptionOfTx2Only := rc2.Difference(rc1)
```

Reply:
I see, so this is not for different blocks,    but for a building block

Reply:
one thing I'm confused about, if using `// skip tx2 if rowConsumptionOfTx2Only.IsOverflown() = true`, shouldn't the Difference function return the `absolute value` instead of `diff value`?